scipy/stats/_continuous_distns.py

```diff
@@ -4346,8 +4346,17 @@ def _stats(self, a, moments='mvsk'):
         return m1, m2, g1, g2
 
     def _entropy(self, a):
-        return a - (a+1.0) * sc.psi(a) + sc.gammaln(a)
+        a = np.asarray([a])
+        def regular(a):
+            h= a - (a+1.0) * sc.psi(a) + sc.gammaln(a)
+            return h
+
+        def asymptotic(a):
+            h = -1.5 * np.log(a) + 0.5 * np.log(2 * np.pi) + 0.5 + 1 / (2 * a)
+            return h
 
+        h = _lazywhere(a >= 1e7, (a), f=asymptotic, f2=regular)
```

Comment:
```suggestion
        h = _lazywhere(a >= 8e6, (a), f=asymptotic, f2=regular)
```

Juding from the [plot](https://github.com/scipy/scipy/pull/18122#issuecomment-1464501670), this seems appropriate to me as the error of the asymptotic expansion is lower already.

Reply:
No, zooming into the graph, the main formula is definitely more accurate overall than the asymptotic expansion at 8e6.
![Figure_1](https://user-images.githubusercontent.com/33975162/224508511-5f1f1eff-f05b-4335-b6bc-13b5892bfee3.png)

Reply:
Thansk for the detailed plot. Can we agree on the cutoff `8.5e6` instead? From there, the expansion is defintely better.

Reply:
>Can we agree on the cutoff 8.5e6 instead? From there, the expansion is defintely better.

The area of the main error that's underneath the extra term line seems to be significantly larger than the area above it. Maybe it's safe to cutoff at 9e6, but 1e7 seems best to me.

Reply:
Another zoomed in plot
![Figure_1](https://user-images.githubusercontent.com/33975162/224513828-e214d917-5e0d-4372-849c-d990074e558c.png)